File: day3/ans.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        
        std::multiset<char> ransom_note_set = {std::begin(ransomNote), std::end(ransomNote)};
        std::multiset<char> magazine_set = {std::begin(magazine), std::end(magazine)};
        
        std::multiset<char> diff;
        
        std::set_difference(ransom_note_set.begin(), ransom_note_set.end(),
                            magazine_set.begin(), magazine_set.end(),
                            std::inserter(diff, diff.begin()));
        
        return diff.empty();
    }
};

File: day3/ans.py
from collections import defaultdict

class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        ransom_note_map = defaultdict(int)
        for char in ransomNote:
            ransom_note_map[char] += 1
        
        for char in magazine:
            if char in ransom_note_map and ransom_note_map[char] != 0:
                ransom_note_map[char] -= 1
        
        return sum(ransom_note_map.values()) == 0
